src/features/auth/components/AuthGuard.tsx

```diff
@@ -0,0 +1,40 @@
+import { useRouter } from 'next/router';
+import { ReactNode, useEffect, useState } from 'react';
+
+import useAuth from '../hooks/useAuth';
+
+import { toast } from '@src/shared/utils/toast';
+
+interface Props {
+  children: ReactNode;
+}
+
+/**
+ * 인증이 필요한 page접근시 사용
+ */
+const AuthGuard = ({ children }: Props) => {
+  const { authState } = useAuth();
+  const router = useRouter();
+  const [isChecked, setIsChecked] = useState(false);
+
+  useEffect(() => {
+    if (!router.isReady) {
+      setIsChecked(false);
+      return;
+    }
+    if (!authState.isAuthenticated) {
+      toast.error('로그인이 필요합니다.');
+      setIsChecked(false);
+      router.replace('/');
+      return;
+    }
+    setIsChecked(true);
+  }, [router, authState.isAuthenticated]);
+
+  if (!isChecked) {
+    return null;
+  }
+  return <>{children}</>;
```

Comment:
isChecked는 불필요한 상태로 보이는데, isAuthenticated로만 판단하면 어떤 문제가 생기나요??
렌더링 부분은 이렇게 변경하면 될 것 같아요
```suggestion
  if (!isAuthenticated) {
    return null;
  }
  return <>{children}</>;
```

Reply:
그리고 AuthGuard는 기존 컴포넌트 파일 구조와 다른데, 맞춰주시면 좋을 것같습니다~!

Reply:
지금 보니까 불필요한 상태였네요 ㅎㅎ 제거하겠습니다~
authguard구조 맞춰서 푸시하겠습니다!!@!